packages/daemon/src/daemon.js

```diff
@@ -93,33 +78,12 @@ const makeEndoBootstrap = (
    * @param {import('@endo/eventual-send').ERef<AsyncIterableIterator<string>>} readerRef
    */
   const storeReaderRef = async readerRef => {
-    const storageDirectoryPath = powers.joinPath(
-      locator.statePath,
-      'store-sha512',
-    );
-    await powers.makePath(storageDirectoryPath);
-
-    // Pump the reader into a temporary file and hash.
-    // We use a temporary file to avoid leaving a partially writen object,
-    // but also because we won't know the name we will use until we've
-    // completed the hash.
-    const digester = powers.makeSha512();
-    const storageId512 = await powers.randomHex512();
-    const temporaryStoragePath = powers.joinPath(
-      storageDirectoryPath,
-      storageId512,
-    );
-    const writer = powers.makeFileWriter(temporaryStoragePath);
+    const { writer, getSha512Hex } = await powers.makeHashedContentWriter(locator.statePath);
```

Comment:
There should only be one content address store instance for the lifetime of a daemon process.